go/consensus/tendermint/full/full.go

```diff
@@ -1281,10 +1283,17 @@ func (t *fullService) lazyInit() error {
 		return err
 	}
 
+	// Migrate Tendermint databases if requested.
+	if viper.GetBool(CfgMigrateTendermintDB) {
```

Comment:
How?  Tendermint has no way of figuring out if a database needs migration or not, the migration script basically just checks for some keys and renames them if it finds them.  Running this every time is too time-consuming.

As it is now, the user just has to pass an additional flag after upgrading.  There's nothing wrong if the user forgets to remove the flag on the next invocation, the node will just take longer to start up, since it will scan all the keys in the db again.  I think it's a reasonable compromise given the state of things.

Reply:
As part of the migration set a `dbVersion` key or something, to suppress trying to migrate again, get rid of the flag.